flask_resty/testing.py

```diff
@@ -61,15 +61,30 @@ def assert_shape(actual, expected):
         # superset of the expected items, rather than that they match.
         for key, value in expected.items():
             if value is not UNDEFINED:
-                assert key in actual
+                assert (
+                    key in actual
+                ), f"Expected key {key} not found in: {actual}"
+
                 assert_shape(actual[key], value)
             else:
                 assert key not in actual
     elif isinstance(expected, (str, bytes)):
         assert expected == actual
     elif isinstance(expected, Sequence):
-        assert isinstance(actual, Sequence)
-        assert len(actual) == len(expected)
+        assert isinstance(
+            actual, Sequence
+        ), "Received a Sequence but actual value is not a Sequence"
```

Comment:
suggestion: include the `actual` value here

```suggestion
        ), f"{actual!r} is not a Sequence"
```

Reply:
Didn't know about `!r` !, but how come it's desirable? Without it, if the input was a string, it would call `str()`, which might be more desirable from what I understand?

Reply:
`repr`s will generally be more detailed than `str` representations, and for strings, it's more readable to have the quotes

```
In [9]: actual = "number"

In [10]: print(f"{actual} is not a string")
number is not a string

In [11]: print(f"{actual!r} is not a string")
'number' is not a string

```